agent/engine/dockerstate/docker_task_engine_state.go

```diff
@@ -143,6 +146,22 @@ func (state *DockerTaskEngineState) ContainerMapByArn(arn string) (map[string]*a
 	return ret, ok
 }
 
+// TaskByShortID retrieves the task of a given docker short container id
+func (state *DockerTaskEngineState) TaskByShortID(cid string) ([]*api.Task, bool) {
+	state.lock.RLock()
+	defer state.lock.RUnlock()
+
+	var tasks []*api.Task
+	for id := range state.idToTask {
+		if strings.HasPrefix(id, cid) {
+			task, _ := state.TaskByID(id)
```

Comment:
@aaithal happy to make the recommended changes. Do we want to add the `GetAllContainerIDs` to the API def or just make it private? @samuelkarp ?